src/library/scala/collection/immutable/Map.scala

```diff
@@ -136,6 +136,28 @@ object Map extends ImmutableMapFactory[Map] {
     override def hashCode: Int = MurmurHash3.emptyMapHash
     override private[immutable] def foreachEntry[U](f: (Any, Nothing) => U): Unit = ()
   }
+  @SerialVersionUID(3L)
+  private abstract class MapNIterator[K,V](n: Int) extends AbstractIterator[(K,V)] with Serializable {
+    private[this] var current = 0
```

Comment:
seems a bit unnecessary to keep track of current and remainder

Reply:
agreed